packages/playwright-crawler/src/internals/adaptive-playwright-crawler.ts

```diff
@@ -175,7 +183,7 @@ export class AdaptivePlaywrightCrawler extends PlaywrightCrawler {
         override readonly config = Configuration.getGlobalConfig(),
     ) {
         super(options, config);
-        this.adaptiveRequestHandler = requestHandler;
+        this.adaptiveRequestHandler = requestHandler ?? (this.router as any);
```

Comment:
Might as well add a ! here to avoid using it further down the line...

Reply:
looks like we dont even need the `as any` here, so the only lie to the compiler in the end is the router override